klippy/extras/pause_resume.py

```diff
@@ -17,8 +17,29 @@ def __init__(self, config):
         self.gcode.register_command("PAUSE", self.cmd_PAUSE)
         self.gcode.register_command("RESUME", self.cmd_RESUME)
         self.gcode.register_command("CLEAR_PAUSE", self.cmd_CLEAR_PAUSE)
+        webhooks = self.printer.lookup_object('webhooks')
+        webhooks.register_endpoint(
+            "pause_resume/pause", self._handle_web_request,
+            methods=["POST"])
+        webhooks.register_endpoint(
+            "pause_resume/resume", self._handle_web_request,
+            methods=["POST"])
     def handle_ready(self):
         self.v_sd = self.printer.lookup_object('virtual_sdcard', None)
+    def _handle_web_request(self, web_request):
+        path = web_request.get_path()
+        if path == "pause_resume/pause":
+            if self.is_paused:
+                raise web_request.error("Print Already Paused")
+            script = "PAUSE"
+        elif path == "pause_resume/resume":
+            if not self.is_paused:
+                raise web_request.error("Print Not Paused")
+            script = "RESUME"
+        else:
+            raise web_request.error("Invalid Path")
+        web_request.put('script', script)
+        self.gcode.run_script_from_remote(web_request)
```

Comment:
Similar to previous comments, is there a need for this endpoint vs using the regular gcode/script endpoint?  Is it for the error handling?

Reply:
As with restart and firmware restart, the primary reason for these is visibility.  It is acceptable to simply use `gcode/script` for them.

Reply:
Okay.  FYI, one thing to aware of in the above is that self.is_paused may change between the time the _handle_web_request() is called and when the gcode "PAUSE" script is run.